airflow/utils/log/task_handler_with_custom_formatter.py

```diff
@@ -45,7 +45,7 @@ def set_context(self, ti) -> None:
         :param ti:
         :return:
         """
-        if ti.raw or self.formatter is None:
+        if ti.raw or self.formatter is None or self.prefix_jinja_template is not None:
```

Comment:
Not sure whether we should add a comment on why `self.prefix_jinja_template is not None:` is added. It might not be that straightforward when looking at this condition alone. Also, could we also add a test case for it?

Reply:
I just take another look and it looks good to me now. Thanks!